src/System.Windows.Forms/src/System/Windows/Forms/ListView.cs

```diff
@@ -2256,13 +2256,12 @@ internal unsafe void AnnounceColumnHeader(Point point)
         private void ApplyUpdateCachedItems()
         {
             // first check if there is a delayed update array
-            ArrayList? newItems = (ArrayList?)Properties.GetObject(PropDelayedUpdateItems);
+            List<ListViewItem>? newItems = (List<ListViewItem>?)Properties.GetObject(PropDelayedUpdateItems);
             if (newItems is not null)
             {
                 // if there is, clear it and push the items in.
-                //
                 Properties.SetObject(PropDelayedUpdateItems, null);
-                ListViewItem[] items = (ListViewItem[])newItems.ToArray(typeof(ListViewItem));
+                ListViewItem[] items = newItems.ToArray();
                 if (items.Length > 0)
                 {
                     InsertItems(_itemCount, items, false /*checkHosting*/);
```

Comment:
Just a nit, but we can avoid a useless allocation here
```suggestion
                if (newItems.Count > 0)
                {
                    InsertItems(_itemCount, newItems.ToArray(), false /*checkHosting*/);
```

Reply:
another nit: `checkHosting : false` instead of the comment